backend/src/attraction/repositories/attraction_repo.py

```diff
@@ -14,6 +21,45 @@ class AttractionRepository(Repository):
     def get_all(self, db: Session) -> list[Attraction]:
         return db.query(Attraction).all()
 
+    def get_filtered(self, db: Session, filters: AttractionFilters) -> list[Attraction]:
+        query = db.query(Attraction)
+
+        if filters.city:
+            query = query.filter(Attraction.city == filters.city)
+        if filters.country:
+            query = query.filter(Attraction.country == filters.country)
+        if filters.category:
+            query = query.filter(Attraction.category == filters.category)
+
+        if (
+            filters.radius is not None
+            and filters.longitude is not None
+            and filters.latitude is not None
+        ):
+            point = func.ST_SetSRID(
+                func.ST_MakePoint(filters.longitude, filters.latitude), 4326
+            ).cast(Geography)
+            attraction_point = func.ST_SetSRID(
+                func.ST_MakePoint(Attraction.longitude, Attraction.latitude), 4326
+            ).cast(Geography)
+            query = query.filter(
+                func.ST_DWithin(attraction_point, point, filters.radius)
+            )
+
+        # sorting
+        if filters.sort_by == AttractionSortedBy.TopRated:
```

Comment:
Maybe we can do this in better way ? For example with use of dictionary ? for example 
```

    filter_operations = {
        'city': lambda value: Attraction.city == value,
        'country': lambda value: Attraction.country == value,
        'category': lambda value: Attraction.category == value,
    }

   
    for filter_name, operation in filter_operations.items():
        filter_value = getattr(filters, filter_name, None)
        if filter_value:
            query = query.filter(operation(filter_value))
```